### PR TITLE
Fixed more flaky tests

### PR DESCRIPTION
**Pull request by mssola**
Signed-off-by: Miquel Sabaté Solà <msabate@suse.com>

### CODE REVIEW

**Comment by vitoravelino** [comment]
On average I guess the tests are way faster but I'm still not sure about the stability of rebooting expectation and click_when_enabled helper. 

I agree on reverting the click when enabled change of but not the rebooting text.

**Comment by mssola** [comment]
@vitoravelino I think this PR is fine as it is now. It's way faster, and unit tests passed. I'll trigger more rebuilds, but I think it should good to go. We can improve tests in another PR if you want, but the top priority was to fix the flaky tests...

**Comment by vitoravelino** [comment]
I re-triggered a few times yesterday, the first two had 1 failure and the other 3 passed. I've just re-triggered. It's not so flaky, eventually with 1 or 2 re-triggers pass. If someone feels annoyed we can get back to this.

File: spec/support/helpers.rb [hunk 1]
```diff
@@ -20,8 +20,14 @@ def disabled?(selector)
...
   def click_on_when_enabled(selector)
-    # will wait until it becomes enabled
-    find("#{selector}:not([disabled])", match: :first).click
+    wait_until { have_css("#{selector}:not([disabled])") }
+    find(selector, match: :first).click
+  end
+
+  def wait_until_having_content(text)
+    has_content = false
+    wait_until { has_content = have_content(text) }
```
Comment (mssola) [inline spec/support/helpers.rb]: Are you sure this doesn't return always true ?

Comment (vitoravelino) [inline spec/support/helpers.rb]: You mean the `wait_until`?

**Comment by vitoravelino** [comment]
After I added the wait_until, duration increased a lot. I ran a few times and not sure if it's just a coincidence. :(

File: spec/support/helpers.rb (hunk 1 above)
Comment (mssola) [inline spec/support/helpers.rb]: The block inside of it.

Comment (vitoravelino) [inline spec/support/helpers.rb]: Yeah, you are right. Need to use the `page.has_text?` as before. Totally forgot.

**Comment by vitoravelino** [comment]
@mssola any other suggestion? If not, feel free to squash everything.

**Comment by mssola** [comment]
This approach is taking us nowhere. We'll start re-visiting how we mock the backend, since maybe it's stuck on some heavy-lifting from that side...

**Comment by mssola** [comment]
Rebased on top of master to gain all the changes on the CI

**Comment by mssola** [comment]
Pushed the branch into the upstream repository so it can be picked up by another PR.

**Pull request closed without merging**